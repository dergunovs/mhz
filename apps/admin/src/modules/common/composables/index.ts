import { useRoute } from 'vue-router';
import { useQueryClient } from '@tanstack/vue-query';

const queryClient = useQueryClient(); // стоит проверить, вызвав invalidate

export function useInvalidate(queryKey: string) {
  const route = useRoute(); 
  function invalidate() {
    if (route.query.create === '1') {
      queryClient.invalidateQueries({ queryKey: [queryKey], exact: true });
    }
  }

  return {
    invalidate,
  };
}
