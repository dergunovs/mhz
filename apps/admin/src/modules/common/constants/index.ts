export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
