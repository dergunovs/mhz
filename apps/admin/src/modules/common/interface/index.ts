import { ICategory, IManager, IManufacturer, IProduct } from 'mhz-types';

export interface ISearchResult {
  products: IProduct[];
  categories: ICategory[];
  manufacturers: IManufacturer[];
  managers: IManager[];
}
