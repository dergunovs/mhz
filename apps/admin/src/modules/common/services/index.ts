import { Ref } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import {
  API_UPLOAD,
  API_UPLOAD_SINGLE,
  API_SEARCH,
  API_STATS_COUNT,
  ISearchResults,
  IEntitiesCount,
  IBaseReply,
} from 'mhz-contracts';

export function search(query: Ref<string>, isAdmin?: boolean) {
  async function fn() {
    const { data } = await api.get<ISearchResults>(API_SEARCH, { params: { search: query.value, isAdmin } });

    return data;
  }

  return useQuery({ queryKey: [API_SEARCH, query], queryFn: fn, enabled: false });
}

export function getEntitiesCount() {
  async function fn() {
    const { data } = await api.get<IEntitiesCount>(API_STATS_COUNT);

    return data;
  }

  return useQuery({ queryKey: [API_STATS_COUNT], queryFn: fn });
}

export function uploadFile(options: object, width?: string) {
  async function fn(file?: File) {
    if (!file) throw new Error();

    const formData = new FormData();

    formData.append('file', file);

    const { data } = await api.post<string>(API_UPLOAD_SINGLE, formData, { params: { width } });

    return data;
  }

  return useMutation({ mutationKey: [API_UPLOAD], mutationFn: fn, ...options });
}

export function uploadFiles(options: object, width?: string, thumb?: boolean) {
  async function fn(files: File[]) {
    const formData = new FormData();

    files.forEach((file) => formData.append('files', file));

    const { data } = await api.post<string[]>(API_UPLOAD, formData, { params: { width, thumb } });

    return data;
  }

  return useMutation({ mutationKey: [API_UPLOAD], mutationFn: fn, ...options });
}

export function deleteFile(thumb?: boolean) {
  async function fn(filename: string) {
    const { data } = await api.delete<IBaseReply>(`${API_UPLOAD}/${filename}`, { params: { thumb } });

    return data;
  }

  return useMutation({ mutationKey: [API_UPLOAD], mutationFn: fn });
}
