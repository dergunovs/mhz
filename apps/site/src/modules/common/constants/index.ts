export const URL_MAIN = '/';
export const URL_ERROR = '/404';

export const PATH_UPLOAD = import.meta.env.VITE_PATH_UPLOAD;
