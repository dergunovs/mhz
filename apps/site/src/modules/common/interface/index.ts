import { IEntity } from 'mhz-types';

export interface ISearchResults {
  [key: string]: IEntity[];
}
