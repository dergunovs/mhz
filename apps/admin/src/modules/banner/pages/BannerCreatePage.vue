<template>
  <div>
    <PageTitle :links="links">{{ title }}</PageTitle>

    <BannerForm />
  </div>
</template>

<script setup lang="ts">
import { useHead } from '@unhead/vue';

import PageTitle from '@/layout/components/PageTitle.vue';
import BannerForm from '@/banner/components/BannerForm.vue';

import { URL_BANNER, URL_BANNER_CREATE } from '@/banner/constants';
import { URL_MAIN } from '@/common/constants';

const title = 'Add banner';

const links = [
  { url: URL_MAIN, title: 'Main' },
  { url: URL_BANNER, title: 'Banners' },
  { url: URL_BANNER_CREATE, title },
];

useHead({
  title,
});
</script>
