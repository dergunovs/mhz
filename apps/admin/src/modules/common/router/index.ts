import { createRouter, createWebHistory } from 'vue-router';

import { routes } from '@/common/router/routes';
import { getCookieToken } from '@/auth/composables';
import { URL_LOGIN } from '@/auth/constants';

const router = createRouter({
  history: createWebHistory(),
  routes,
  scrollBehavior(to, from, savedPosition) {
    return savedPosition ? savedPosition : { top: 0 };
  },
});

router.beforeEach((to, from, next) => {
  if (to.path !== URL_LOGIN && !getCookieToken()) {
    next({ path: `${URL_LOGIN}?logout=1` });
  } else {
    next();
  }
});

export default router;
