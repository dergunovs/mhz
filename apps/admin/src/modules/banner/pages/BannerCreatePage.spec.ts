import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import BannerCreatePage from './BannerCreatePage.vue';

import { wrapperFactory } from '@/common/test';

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(BannerCreatePage, {});
});

enableAutoUnmount(afterEach);

describe('BannerCreatePage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(BannerCreatePage)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
