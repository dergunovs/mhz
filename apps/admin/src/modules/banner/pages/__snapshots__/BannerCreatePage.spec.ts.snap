// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BannerCreatePage > matches snapshot 1`] = `
"<div>
  <page-title-stub links="[object Object],[object Object],[object Object]"></page-title-stub>
  <banner-form-stub></banner-form-stub>
</div>"
`;
